magnum-kyo/src/main/scala/com/augustnagro/magnum/magkyo/TransactorKyo.scala

```diff
@@ -0,0 +1,201 @@
+package com.augustnagro.magnum.magkyo
+
+import com.augustnagro.magnum.{DbCon, DbTx, SqlException, SqlLogger}
+import kyo.*
+
+import java.sql.Connection
+import javax.sql.DataSource
+import scala.util.control.NonFatal
+
+class TransactorKyo private (
+    dataSource: DataSource,
+    sqlLogger: SqlLogger,
+    connectionConfig: Connection => Unit,
+    semaphore: Maybe[Meter]
+):
+  def withSqlLogger(sqlLogger: SqlLogger): TransactorKyo =
+    new TransactorKyo(
+      dataSource,
+      sqlLogger,
+      connectionConfig,
+      semaphore
+    )
+
+  def withConnectionConfig(
+      connectionConfig: Connection => Unit
+  ): TransactorKyo =
+    new TransactorKyo(
+      dataSource,
+      sqlLogger,
+      connectionConfig,
+      semaphore
+    )
+
+  def connect[A](f: DbCon ?=> A): A < (Abort[Throwable] & Async) =
+    val res =
+      Resource
+        .acquireRelease(acquireConnection)(releaseConnection)
+        .map { lo =>
+          IO {
+            connectionConfig(lo)
+            f(using DbCon(lo, sqlLogger))
+          }
+        }
+    val effect = Resource.run(res)
```

Comment:
I think you shouldn't need a `Resource` given you fulfill it right away.
```suggestion
    val effect =
      IO.ensure(releaseConnection)(
      	acquireConnection
        .map { lo =>
          IO {
            connectionConfig(lo)
            f(using DbCon(lo, sqlLogger))
          }
        }
```

Reply:
Sorry I just realized we can't do this - I believe we need a slightly more powerful operator (less powerful than Resource perhaps?). We need a [bracket](https://javadoc.io/doc/dev.zio/zio_3/latest/zio/ZIO$.html#acquireRelease-f6a) operator on IO.